robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java

```diff
@@ -0,0 +1,241 @@
+package com.team2357.frc2022.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.team2357.frc2022.Constants;
+import com.team2357.frc2022.util.VisionTargetSupplier;
+import com.team2357.lib.arduino.ArduinoUSBController;
+import com.team2357.lib.subsystems.ClosedLoopSubsystem;
+import com.team2357.lib.subsystems.LimelightSubsystem.VisionTarget;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class TurretSubsystem extends ClosedLoopSubsystem {
+    CANSparkMax m_turretMotor;
+    private SparkMaxPIDController m_pidController;
+    private ArduinoUSBController m_arduinoHallEffectSensor;
+    Configuration m_config;
+
+    private VisionTargetSupplier m_targetSupplier;
+    private VisionTarget m_currentTarget;
+
+    private double m_setPoint;
```

Comment:
I'm thinking we may want to use actual degrees of rotation for turret position, with zero pointing straight ahead (centered on the intake, I would guess) We could use `m_positionDegrees` for both our current position, and if we're zeroed or not, by setting the value to NaN or a specific constant if calibration failed.

We may want to keep a separate current position and a target position as well, for when we're transitioning between poses.